src/common/accountParser.ts

```diff
@@ -180,39 +181,67 @@ const TokenAccountLayout = BL.struct([
   pubkeyField("closeAuthority")
 ])
 
-export const parseTokenAccount = (account: Buffer, accountPubkey: PublicKey) => {
-  const data = TokenAccountLayout.decode(account)
+/** Buffer layout for de/serializing a mint */
+export const RawMintLayout = BL.struct<RawMint>([
+  BL.u32("mintAuthorityOption"),
+  BLU.publicKey("mintAuthority"),
+  u64Field("supply"),
+  BL.u8("decimals"),
+  BLU.bool("isInitialized"),
+  BL.u32("freezeAuthorityOption"),
+  BLU.publicKey("freezeAuthority")
+])
+
+/**
+ * Decode a token account
+ * @param {Buffer} account
+ * @param {PublicKey} accountPubkey
+ * @returns
+ */
+export const parseTokenAccount = (account: Buffer, accountAddress: PublicKey): TokenAccountInfo => {
+  const data = RawTokenAccountLayout.decode(account)
 
   // PublicKeys and BNs are currently Uint8 arrays and
   // booleans are really Uint8s. Convert them
   const decoded: TokenAccountInfo = {
-    address: accountPubkey,
+    address: accountAddress,
     mint: new PublicKey(data.mint),
     owner: new PublicKey(data.owner),
     amount: new BN(data.amount, undefined, "le"),
-    delegate: (data as any).delegateOption ? new PublicKey(data.delegate) : null,
+    delegate: new PublicKey(data.delegate),
```

Comment:
If delegateOption === 1 then create the public key, any other value, the public key should be undefined. Adjust the interface as necessary
```suggestion
    delegate: data.delegateOption === 1 ? new PublicKey(data.delegate) : undefined,
```